lib/solarwinds_apm/opentelemetry/solarwinds_sampler.rb

```diff
@@ -111,33 +111,40 @@ def calculate_liboboe_decision(parent_span_context, xtraceoptions, name, kind, a
           SolarWindsAPM::TransactionCache.set(transaction_naming_key, tracing_mode)
         end
 
-        sw_member_value    = parent_span_context.tracestate[SolarWindsAPM::Constants::INTL_SWO_TRACESTATE_KEY]
-        trigger_trace_mode = SolarWindsAPM::OboeTracingMode.get_oboe_trigger_trace_mode(@config['trigger_trace'])
-        sample_rate        = UNSET
-        options            = xtraceoptions&.options
-        trigger_trace      = xtraceoptions&.intify_trigger_trace || 0
-        signature          = xtraceoptions&.signature
-        timestamp          = xtraceoptions&.timestamp
+        if SolarWindsAPM.is_lambda
+          # trigger trace disabled at this point (https://swicloud.atlassian.net/wiki/spaces/NIT/pages/3753116438/AWS+Lambda+Instrumentation+POC#Concerns)
+          do_metrics, do_sample, rate, source, bucket_rate,
+            bucket_cap, decision_type, auth, status_msg, auth_msg,
+              status = SolarWindsAPM.oboe_api.getTracingDecision
```

Comment:
True, we don't need to support trigger trace for Lambda (yet), but this also seems to skip setting the non-trigger trace args to `getTracingDecisions`, esp. incoming trace context. I think the [following args](https://github.com/solarwinds-cloud/solarwinds-apm-liboboe/blob/master/liboboe/oboe_api.h#L553-L555) still need to be passed to get decisions:
* tracestring
* sw_member_value
* tracing_mode

Reply:
In fact there's probably no harm to let all the args be set even for lambda. An unsigned trigger trace request would still work, the signed TT and the injection of response `x-trace` would not which is current known gap with full TT support.

Reply:
updated!